### PR TITLE
feat: docker compose env

### PR DESCRIPTION
**Pull request by leopaul36**
This PR sets up required files and instructions to create a local Docker-based Avalanche local network and create a Nodekit SEQ Subnet / Blockchain on it.

Everything is documented in `docker/README.md`

As mentioned in the docs, the setup **works** but this [bug](https://github.com/ava-labs/hypersdk/issues/592) still occurs **upon restart** of the Avalanche nodes containers which makes it far from ideal. It is a limitation of HyperSDK that we can not go around at this time (the issue is not related to Nodekit SEQ and also occurs with the vanilla tokenvm binary).

To make things easier for local development, we have created a special [`nodekit-seq`](https://github.com/AshAvalanche/ansible-avalanche-getting-started/tree/nodekit-seq) branch in our [`ansible-avalanche-getting-started`](https://github.com/AshAvalanche/ansible-avalanche-getting-started) repository with a working setup and documentation (see instructions in `README.md` for local development of Nodekit SEQ in a VM-based environment. It is much more suitable for development because it allows for:
- AvalancheGo restarts
- Nodekit SEQ binary upgrades
- Uses production-ready Ansible Collection. Moving from this env. to a production env. will be seamless.

In conclusion, we would not recommend using the Docker-based environment at this time for anything other than short-lived SEQ Subnet local testing that do not require AvalancheGo restart and / or VM upgrade. We can re-evaluate this later when / if HyperSDK shows code improvements that would fix the issues we encounter now.

### CODE REVIEW

**Comment by bianyuanop** [comment]
Great work! Thanks!
